src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.ts

```diff
@@ -44,6 +48,9 @@ export class CodeEditorStudentContainerComponent implements OnInit, OnDestroy {
     exercise: ProgrammingExercise;
     course?: Course;
 
+    activatedExerciseHints: ExerciseHint[];
+    availableExerciseHints: ExerciseHint[];
```

Comment:
I think this could lead to problems otherwise, the attibute is expected to never be undefined, but it will be if the server endpoint throws an excetption.
```suggestion
    activatedExerciseHints: ExerciseHint[] = [];
    availableExerciseHints: ExerciseHint[] =  [];
```

Reply:
The button is only accessible if both lists are defined (`exercise-hint-button-overlay.component.ts`):

`*ngIf="availableExerciseHints && activatedExerciseHints && availableExerciseHints.length + activatedExerciseHints.length > 0"`

Reply:
ok, but then the type should be `activatedExerciseHints?: ExerciseHint[]`

Reply:
You're right. I implemented it in 0ef6b30.